pennylane/ops/functions/is_commuting.py

```diff
@@ -374,165 +324,55 @@ def is_commuting(operation1, operation2):
-
-        target_wires_2 = qml.wires.Wires(
-            [w for w in operation2.wires if w not in operation2.control_wires]
-        )
-
-        target_control = intersection(operation1.wires, operation2.control_wires)
-        target_target = intersection(operation1.wires, target_wires_2)
-
-        # Case 4.1: control and target 2 overlap with target 1
-        if target_control and target_target:
-            return _commutes(control_base_2, operation1.name)
-
-        # Case 4.2: control operation 2 overlap with target 1
-        if target_control and not target_target:
-            return _commutes("ctrl", operation1.name)
-
-        # Case 4.3: target 1 overlaps with target 2
-        if not target_control and target_target:
-            return _commutes(control_base_2, operation1.name)
-
-    # Case 5: no controlled operations
-    # Case 5.1: no controlled operations we simply check the commutation table
-    return _commutes(operation1.name, operation2.name)
-
-
-def _both_controlled(control_base, operation1, operation2):
-    """Auxiliary function to the is_commuting function for the case when both
-    operations are controlled."""
-    # pylint: disable=too-many-branches
-    # pylint: disable=too-many-return-statements
-
-    if control_base.get(operation1.name) != "ControlledOperation":
-        control_base_1 = control_base.get(operation1.name)
-    else:
-        control_base_1 = operation1.control_base
-
-    if control_base.get(operation2.name) != "ControlledOperation":
-        control_base_2 = control_base.get(operation2.name)
-    else:
-        control_base_2 = operation2.control_base
-
-    target_wires_1 = qml.wires.Wires(
-        [w for w in operation1.wires if w not in operation1.control_wires]
-    )
-    target_wires_2 = qml.wires.Wires(
-        [w for w in operation2.wires if w not in operation2.control_wires]
-    )
-
-    control_control = intersection(operation1.control_wires, operation2.control_wires)
-    target_target = intersection(target_wires_1, target_wires_2)
-    control_target = intersection(operation1.control_wires, target_wires_2)
-    target_control = intersection(target_wires_1, operation2.control_wires)
-
-    # Case 2.1: disjoint targets
-    if control_control and not target_target and not control_target and not target_control:
-        return True
-
-    # Case 2.2: disjoint controls
-    if not control_control and target_target and not control_target and not target_control:
-        return _commutes(control_base_2, control_base_1)
+    op_set = {"U2", "U3", "Rot", "CRot"}
+    if operation1.name in op_set and operation2.name in op_set:
+        return check_commutation_two_non_simplified_rotations(operation1, operation2)
 
-    # Case 2.3: targets overlap and controls overlap
-    if target_target and control_control and not control_target and not target_control:
-        return _commutes(control_base_1, control_base_2)
+    control_base_1 = _get_target_name(operation1)
```

Comment:
We now treat the cases of:
1) both controlled
2) one controlled and the other not
3) neither controlled

All the same.  We may just get empty overlaps with control wires if one operation is not controlled.  This greatly streamlines the logical complexity, allowing us to condense all potential cases into three checks.